### PR TITLE
mat: add SymBanded type

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

### CODE REVIEW

File: mat/symband.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ var (
 	_            Matrix           = symBandDense
 	_            Symmetric        = symBandDense
 	_            Banded           = symBandDense
+	_            SymBanded        = symBandDense
```
Comment (btracey) [inline mat/symband.go]: Diagonal should also have the methods added to satisfy this

File: mat/symband.go [hunk 2]
```diff
@@ -27,6 +28,22 @@ type SymBandDense struct {
 	mat blas64.SymmetricBand
 }
 
+// SymBanded is a symmetric band matrix interface type.
+type SymBanded interface {
```
Comment (btracey) [inline mat/symband.go]: Wouldn't this be better expressed as `Banded` and then the extra `Symmetric`?

Comment (btracey) [reply mat/symband.go]: (in terms of the expression of the interface methods)

File: mat/symband.go (hunk 1 above)
Comment (kortschak) [reply mat/symband.go]: I think that's another PR.

File: mat/symband.go (hunk 2 above)
Comment (kortschak) [reply mat/symband.go]: I couldn't decide which way that should go, so I avoided it. I guess this way is the lesser number of methods literally present. I'll change it.

Comment (kortschak) [reply mat/symband.go]: Though it's at odds with how the mutable type is expressed.

Comment (btracey) [reply mat/symband.go]: Is it? MutableBanded is Banded plus a Mutable method

Comment (kortschak) [reply mat/symband.go]: `MuatbleSymBanded` is `Symmetric`+some methods. This is `Banded`+some methods.

**Pull request merged**